### PR TITLE
[Network] Partial bindings of Network for Xcode 12 beta 6.

### PR DESCRIPTION
**Pull request by mandel-macaque**
Bound the NWConnectionGroup and all its methods and enums. Tests cannot
be added until NWGroupDescriptor is bound. All tests for this class and
NWGroupDescriptor will be added in the following PR.

### CODE REVIEW

**Comment by mandel-macaque** [comment]
Note to reviewers, I'm using tabs, no idea why github decides to use two spaces...

File: src/Network/NWConnectionGroup.cs
Comment (tj-devel709) [inline src/Network/NWConnectionGroup.cs:171]: To my understanding, a question mark denotes a nullable type, but since you are not declaring 'del' here, why is there a question mark?

**Review by tj-devel709 — APPROVED** [review]
Everything looks good to me!

File: src/Network/NWConnectionGroup.cs
Comment (mandel-macaque) [reply src/Network/NWConnectionGroup.cs:171]: In this case is because error can be null, the ? will mean, if not null, call the method, else don't

File: src/Network/NWConnectionGroup.cs
Comment (rolfbjarne) [inline src/Network/NWConnectionGroup.cs:48]: Maybe use `GetCheckedHandle` here?

File: src/Network/NWConnectionGroup.cs
Comment (rolfbjarne) [inline src/Network/NWConnectionGroup.cs:154]: `content.GetHandle ()` is a safe way to get the Handle or IntPtr.Zero.

File: src/Network/NWConnectionGroup.cs
Comment (rolfbjarne) [inline src/Network/NWConnectionGroup.cs:182]:
```suggestion
						content.GetHandle (),
						endpoint.GetHandle (),
```

This can be done in many other places as well.

File: src/Network/NWConnectionGroup.cs
Comment (rolfbjarne) [inline src/Network/NWConnectionGroup.cs:183]: GetCheckedHandle?

```suggestion
						context.GetCheckedHandle (),
```

This can be done in many other places as well.

File: src/Network/NWConnectionGroup.cs
Comment (rolfbjarne) [inline src/Network/NWConnectionGroup.cs:214]: This way you wouldn't need to manually Dispose later:

```suggestion
				using var nsContent = new DispatchData (content, owns: false);
				using var nsContext = new NWContentContext (context, owns: false);
```

File: src/Network/NWConnectionGroup.cs
Comment (rolfbjarne) [inline src/Network/NWConnectionGroup.cs:217]: And that would also fix a bug, because you're Disposing the same instance twice here:

```suggestion
```

File: src/Network/NWConnectionGroup.cs
Comment (rolfbjarne) [inline src/Network/NWConnectionGroup.cs:251]: Does adding `using` work here?

```suggestion
				using var nwError = (error == IntPtr.Zero) ? null : new NWError (error, owns: false);
```

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

File: src/Network/NWConnectionGroup.cs
Comment (mandel-macaque) [reply src/Network/NWConnectionGroup.cs:217]: Good catch. Stupid autocomplete 🤦🏼‍♂️

File: src/Network/NWConnectionGroup.cs
Comment (spouliot) [inline src/Network/NWConnectionGroup.cs:36]: `protected` (or `protected internal`)

File: src/Network/NWConnectionGroup.cs
Comment (spouliot) [inline src/Network/NWConnectionGroup.cs:140]: Use `GetCheckedHandle` so you don't end up providing a non-null managed instance that is disposed.

IOW when the instance is disposed then `Handle` is `IntPtr.Zero`. You would effectively call the native API with `nil`, which is not allowed.

File: src/Network/NWConnectionGroup.cs
Comment (spouliot) [inline src/Network/NWConnectionGroup.cs:155]: same as earlier, use `GetCheckedHandle`

File: src/Network/NWConnectionGroup.cs
Comment (spouliot) [reply src/Network/NWConnectionGroup.cs:214]: and they would still be disposed if `del` throws :)

File: src/Network/NWGroupDescriptor.cs
Comment (spouliot) [inline src/Network/NWGroupDescriptor.cs:13]: `protected internal` since it seems used outside the type

**Review by spouliot — CHANGES_REQUESTED** [review]
Also needs tests.

Splitting the PR _is_ fine, adding tests later _can_ be fine.

OTOH merging this PR without tests is not since we could release before the tests are ready/merged (and would need extra work to obsolete API).

**Comment by mandel-macaque** [comment]
Do not merge until we have the GM. Certains things have been removed from mac OS X.

File: tests/monotouch-test/Network/NWConnectionGroupTest.cs
Comment (rolfbjarne) [inline tests/monotouch-test/Network/NWConnectionGroupTest.cs:16]:
```suggestion
	// communications between bonjour services, which is very fragile in the CI, a sample 
```

File: src/Network/NWConnectionGroup.cs
Comment (rolfbjarne) [inline src/Network/NWConnectionGroup.cs:154]:
```suggestion
			nw_connection_group_reply (GetCheckedHandle (), inboundMessage.GetCheckedHandle  (), outboundMessage.GetCheckedHandle (), content.GetHandle ());
```

**Review by rolfbjarne — APPROVED** [review]

**Review by spouliot — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

**Comment by mandel-macaque** [comment]
Failure is unrelated with the PR.

**Pull request merged**
